apps/gateway/src/routes/api/v2/projects/[projectId]/versions/[versionUuid]/documents/handlers/run.test.ts

```diff
@@ -0,0 +1,376 @@
+      const document = await createDocumentVersion({
+        workspace,
+        user,
+        commit: cmt,
+        path,
+        content: helpers.createPrompt({ provider: providers[0]! }),
+      })
+
+      commit = await mergeCommit(cmt).then((r) => r.unwrap())
+
+      route = `/api/v2/projects/${project!.id}/versions/${commit!.uuid}/documents/run`
+      body = JSON.stringify({
+        path: document.documentVersion.path,
+        parameters: {},
+      })
+      headers = {
+        Authorization: `Bearer ${token}`,
+        'Content-Type': 'application/json',
+      }
+    })
+
+    it('uses source from __internal', async () => {
+      const stream = new ReadableStream({
+        start(controller) {
+          controller.enqueue({
+            event: StreamEventTypes.Latitude,
+            data: {
+              type: ChainEventTypes.Complete,
+              response: {
+                text: 'Hello',
+                usage: {},
+              },
+            },
+          })
+
+          controller.close()
+        },
+      })
+
+      const response = new Promise((resolve) => {
+        resolve({ text: 'Hello', usage: {} })
+      })
+
+      mocks.runDocumentAtCommit.mockReturnValue(
+        new Promise((resolve) => {
+          resolve(
+            Result.ok({
+              stream,
+              response,
+            }),
+          )
+        }),
+      )
+
+      await app.request(route, {
+        method: 'POST',
+        body: JSON.stringify({
+          path: '/path/to/document',
+          parameters: {},
+          __internal: { source: LogSources.Playground },
+        }),
+        headers,
+      })
+
+      expect(mocks.runDocumentAtCommit).toHaveBeenCalledWith({
+        workspace,
+        document: expect.anything(),
+        commit,
+        parameters: {},
+        source: LogSources.Playground,
+      })
+    })
+
+    it('returns last response', async () => {
+      const stream = new ReadableStream({
+        start(controller) {
+          controller.enqueue({
+            event: StreamEventTypes.Latitude,
+            data: {
+              type: ChainEventTypes.Complete,
+              response: {
+                text: 'Hello',
+                usage: {},
+              },
+            },
+          })
+
+          controller.close()
+        },
+      })
+
+      const response = new Promise((resolve) => {
+        resolve({ text: 'Hello', usage: {} })
+      })
+
+      mocks.runDocumentAtCommit.mockReturnValue(
+        new Promise((resolve) => {
+          resolve(
+            Result.ok({
+              stream,
+              response,
+            }),
+          )
+        }),
+      )
+
+      const res = await app.request(route, {
+        method: 'POST',
+        body,
+        headers,
+      })
+
+      expect(res.status).toBe(200)
+      expect(await res.json()).toEqual({
+        type: ChainEventTypes.Complete,
+        response: {
+          text: 'Hello',
+          usage: {},
+        },
```

Comment:
the complete response should also contain all the conversation messages

Reply:
It does, it contains `messages` and `response`, that is just what was included in the mocked response.